### PR TITLE
fix(SplashScreen): Enable threaded renderer

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do
Part of #9558

Motivation: The splash screen is freezing when the main thread is blocked even with the new animator introduced here #9524.

Enable threaded renderer except when using Qt5 for `arm64` on Macs.

I'm adding the nimqml changes to be reviewed here first. Will add nimqml PR and bump the version when this PR is reviewed.

### Affected areas
Qt renderer

### CODE REVIEW

File: vendor/DOtherSide/lib/src/DOtherSide.cpp [hunk 1]
```diff
@@ -188,6 +189,16 @@ void dos_qguiapplication_initialize_opengl()
     QGuiApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
 }
 
+void dos_qguiapplication_try_enable_threaded_renderer()
+{
+    if(QSysInfo::buildCpuArchitecture() == "arm64" && QSysInfo::kernelType() == "darwin" && QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
```
Comment (caybro) [inline vendor/DOtherSide/lib/src/DOtherSide.cpp:194]:
```suggestion
    if(QSysInfo::currentCpuArchitecture() == "arm64" && QSysInfo::productType() == "osx" && QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
```

**Review by caybro — CHANGES_REQUESTED** [review]
The _build_ architecture & OS (most likely our CI in this case) doesn't matter; what matters though is the runtime environment

**Review by caybro — APPROVED** [review]
LGTM

After confirming the runtime results

**Review by alexandraB99 — APPROVED** [review]

**Comment by alexjba** [comment]
@glitchminer can you please test it on windows?
First please remove `QSG_RENDER_LOOP` env variable if you have set it before.

There are two things we need to see:
1. The splash screen should run smoothly with no freeze
2. The memory usage should be as usual. Please also let the app for min 1h in welcome screen with the password cursor flickering and the PC awake. Keeping the app in this state should isolate the renderer memory usage. After 1h in this state the app should not use more than 1-2mb extra memory.

**Comment by glitchminer** [comment]
Hi @alexjba, I'll do this as soon as possible. I don't have my windows machine available just now so it will likely be tomorrow at the earliest.

**Comment by jrainville** [comment]
I just tested on Windows and it worked great. The animation never stopped

**Comment by glitchminer** [comment]
> I just tested on Windows and it worked great. The animation never stopped

@jrainville , did you have any issues launching the CI build or local build for this? I was discussing this with @alexjba today as attempting to launch the CI build does nothing on my machine. I can launch a previous RC and a local build of master though. 

When attempting to run locally I get this far:

From terminal I get this far:
`Running: bin/nim_status_client.exe
could not import: dos_qguiapplication_try_enable_threaded_renderer
make: *** [Makefile:660: run-windows] Error 1
`

**Comment by jrainville** [comment]
@glitchminer I only tried a local build, and indeed, I had the same error, which is resolved by removing the build of DOtherSide to force its rebuild.

**Review by glitchminer — APPROVED** [review]
Tested, approved.

Animation did not freeze.
App in focus with Welcome screen and cursor waiting for user password:
Memory at start: 265.7
Memory after 2 hours: 262.3
(win11, local build)

@alexjba, just to note the CI build may need looked into as I still couldn't find a way to launch it successfully.

**Comment by glitchminer** [comment]
Hi @jrainville , I believe this is the PR where the CI build did not launch on Windows. Just noting as I understand it was brought up somewhere.

**Comment by alexjba** [comment]
@glitchminer There's a new build available. Could you please give it a try?

**Comment by glitchminer** [comment]
> Ah thanks for letting me know. So CI builds on other PRs work still?

@jrainville, good question! I've not been testing using CIs on Windows regularly but I've attempted with 3 just now. @alexjba, this included the latest one for this PR. None of them showed any signs of starting after installation. I then checked with latest RC 0.10 installation and that did manage to launch.

**Comment by alexjba** [comment]
> > Ah thanks for letting me know. So CI builds on other PRs work still?
> 
> @jrainville, good question! I've not been testing using CIs on Windows regularly but I've attempted with 3 just now. @alexjba, this included the latest one for this PR. None of them showed any signs of starting after installation. I then checked with latest RC 0.10 installation and that did manage to launch.

@glitchminer Thanks for trying it out!
Could you please share the errors you're getting? Would be interesting to see the errors from 2 different PRs.

`Running: bin/nim_status_client.exe could not import: dos_qguiapplication_try_enable_threaded_renderer make: *** [Makefile:660: run-windows] Error 1 `
This error you had initially on this PR is specific to this one, but the other PRs should have other errors. `dos_qguiapplication_try_enable_threaded_renderer ` was introduced here

**Comment by jrainville** [comment]
I'll also be trying CI builds on Windows for other PRs and/or releases. I'll report here

**Comment by jrainville** [comment]
This is latest master:
```
$ ./Status.exe
INFO [03-06|13:50:42.001] Status backend initialized               backend=geth version=0.136.2 commit=8a296407f IpfsGatewayURL=https://ipfs.status.im/
Warning: Ignoring malformed logging rule: '*.debug=true;' (io\qloggingregistry.cpp:257, QLoggingSettingsParser::parseNextLine)
Warning: Ignoring malformed logging rule: 'qt.*.debug=false;' (io\qloggingregistry.cpp:257, QLoggingSettingsParser::parseNextLine)
Warning: Cannot open file 'C:/Users/sur_e/AppData/Local/StatusApp/status.svg', because: The system cannot find the file specified. (:0, )
Warning: Cannot open file 'C:/Users/sur_e/AppData/Local/StatusApp/status.svg', because: The system cannot find the file specified. (:0, )
INF 2023-03-06 13:50:42.285-05:00 Version: 0.9.90                            topics="status-app" tid=11812 file=nim_status_client.nim:169
INF 2023-03-06 13:50:42.285-05:00 Commit: a2a1da801                          topics="status-app" tid=11812 file=nim_status_client.nim:170
INF 2023-03-06 13:50:42.286-05:00 Current date:                              topics="status-app" tid=11812 file=nim_status_client.nim:171 currentDateTime=2023-03-06T13:50:42-05:00
INF 2023-03-06 13:50:42.287-05:00 starting application controller...         topics="status-app" tid=11812 file=nim_status_client.nim:173
Warning: QQmlApplicationEngine failed to load component (:0, )
Warning: qrc:/main.qml:317:9: Type AppMain unavailable (qrc:/main.qml:317, )
Warning: qrc:/app/mainui/AppMain.qml:1:1: module "QtQuick" version 2.15 is not installed (qrc:/app/mainui/AppMain.qml:1, )
INF 2023-03-06 13:50:43.599-05:00 starting application...                    topics="status-app" tid=11812 file=nim_status_client.nim:176
Warning: Error while loading QML: QUrl("qrc:/main.qml") (:0, )
INF 2023-03-06 13:50:43.601-05:00 shutting down...                           topics="status-app" tid=11812 file=nim_status_client.nim:144
INF 2023-03-06 13:50:43.601-05:00 logging out...                             topics="app-controller" tid=11812 file=app_controller.nim:273
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x7ffdeaac9992]

goroutine 17 [running, locked to thread]:
github.com/status-im/status-go/services/ens.(*Service).Stop(...)
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/services/ens/service.go:41
github.com/status-im/status-go/node.(*StatusNode).Cleanup(0x1c0003ae820)
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/node/status_node_services.go:604 +0x112
github.com/status-im/status-go/api.(*GethStatusBackend).cleanupServices(...)
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/api/geth_backend.go:1389
github.com/status-im/status-go/api.(*GethStatusBackend).Logout(0x1c000001860)
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/api/geth_backend.go:1359 +0xb9
github.com/status-im/status-go/mobile.Logout()
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/mobile/status.go:425 +0x25
main.Logout()
        C:/Users/jenkins/workspace/status-desktop/platforms/windows/vendor/status-go/build/bin/statusgo-lib/main.go:305 +0x19
```

**Comment by jrainville** [comment]
Same thing on a PR build. So it has nothing to do with this PR. I'll open an issue

**Comment by alexjba** [comment]
Looks like Windows CI builds are not using Qt 5.15.

**Comment by glitchminer** [comment]
> Same thing on a PR build. So it has nothing to do with this PR. I'll open an issue

Thanks for checking, @jrainville 👍

**Comment by alexjba** [comment]
@glitchminer I've started a new Windows build. This time it's using Qt5.15. Could you please check if it works?
https://ci.status.im/job/status-desktop/job/branches/job/windows/job/PR-9715/7/

**Comment by glitchminer** [comment]
> This time it's using Qt5.15. Could you please check if it works?

@alexjba , good news, that one started ok 🚀👍

**Comment by alexjba** [comment]
> > This time it's using Qt5.15. Could you please check if it works?
> 
> @alexjba , good news, that one started ok 🚀👍

Nice! Thanks! I'll merge it now

**Pull request merged**